### PR TITLE
EJ2-Documentation-876158-Ug-Documentation-For-Editing

### PR DESCRIPTION
**Pull request by harinikrh**

### CODE REVIEW

File: ej2-javascript/grid/editing/edit.md [hunk 1]
```diff
@@ -11,13 +11,15 @@ domainurl: ##DomainURL##
 
 # Edit in ##Platform_Name## Grid control
 
-The Grid component has options to dynamically insert, delete and update records. Editing feature requires a primary key column for CRUD operations. To define the primary key, set [`columns.isPrimaryKey`](../../api/grid/column/#isprimarykey) to `true` in particular column.
+The Grid control provides powerful options for dynamically inserting, deleting, and updating records, enabling you to modify data directly within the grid. This feature is useful when you want to enable you to perform CRUD (Create, Read, Update, Delete) operations seamlessly.
```
Comment (VigneshNatarajan27) [inline ej2-javascript/grid/editing/edit.md:14]: This feature is useful when you want to seamlessly perform CRUD (Create, Read, Update, Delete) operations

File: ej2-javascript/grid/editing/edit.md [hunk 2]
```diff
@@ -113,11 +122,55 @@ In the following demo, editing is disabled for the `CustomerID` column.
 {% previewsample "page.domainurl/code-snippet/grid/grid-cs37" %}
 {% endif %}
 
-## Disable editing for a particular row or cell
+>* If you have set the [isPrimaryKey](../../api/grid/column/#isprimarykey) property to **true** for a column, editing will be automatically disabled for that column.
+> * You can disable the particular row using [actionBegin](../../api/grid/#actionbegin) event.please refer this [link](../../grid/editing/in-line-editing#disable-editing-for-a-particular-row).
```
Comment (VigneshNatarajan27) [inline ej2-javascript/grid/editing/edit.md:126]: either give space or change that dot to comma

File: ej2-javascript/grid/editing/edit.md [hunk 3]
```diff
@@ -175,48 +232,194 @@ For batch mode of editing, you can use [`cellEdit`](../../api/grid/#celledit) ev
...
+## Edit complex column 
+
+The edit template for complex column in Grid is used to customize the editing experience when dealing with complex data structures. This capability is particularly useful for handling nested data objects within grid columns. By default, the grid binds complex data to column fields using the dot (.) operator. However, when you render custom elements, such as input fields, in the edit template for a complex column, you must use the (___) underscore operator instead of the dot (.) operator to bind the complex object.
+
+In the following sample, the input element is rendered in the edit template of the FirstName and LastName column. The edited changes can be saved using the `name` property of the input element. Since the complex data is bound to the FirstName and LastName column, The `name` property should be defined as **Name___FirstName** and **Name___LastName**, respectively, instead of using the dot notation (**Name.FirstName** and **Name.LastName**).
+
+{% if page.publishingplatform == "typescript" %}
+
+ {% tabs %}
+{% highlight ts tabtitle="index.ts" %}
```
Comment (VigneshNatarajan27) [inline ej2-javascript/grid/editing/edit.md:277]: PrimaryKey is not enabled in that Grid. kindly check and resolve.

File: ej2-javascript/grid/editing/edit.md [hunk 4]
```diff
@@ -175,48 +232,194 @@ For batch mode of editing, you can use [`cellEdit`](../../api/grid/#celledit) ev
...
+{% endtabs %}
+
+{% previewsample "page.domainurl/code-snippet/grid/edit-separate-toolbar" %}
+{% endif %}
+
+### Using external form 
+
+Performing the edit operation in a custom external form in the Syncfusion Grid is a valuable feature when you need to customize the edit operation within a separate form rather than the default in-grid editing. 
+
+To enable the use of an external form for editing in Syncfusion Grid, you can make use of the `RowSelected` property. This property specifies whether the edit operation should be triggered when a row is selected.
```
Comment (VigneshNatarajan27) [inline ej2-javascript/grid/editing/edit.md:383]: rowSelected event

it is an event not properly.

change the casing for event.

File: ej2-javascript/grid/editing/edit.md (hunk 1 above)
Comment (harinikrh) [reply ej2-javascript/grid/editing/edit.md:14]: modified this line as "This feature is useful when you want to seamlessly perform CRUD (Create, Read, Update, Delete) operations"

File: ej2-javascript/grid/editing/edit.md [hunk 5]
```diff
@@ -175,48 +232,194 @@ For batch mode of editing, you can use [`cellEdit`](../../api/grid/#celledit) ev
...
 
-{% previewsample "page.domainurl/code-snippet/grid/edit-column-template-cs1" %}
+{% previewsample "page.domainurl/code-snippet/grid/edit-enum-column" %}
+{% endif %}
+
+## Edit complex column 
+
+The edit template for complex column in Grid is used to customize the editing experience when dealing with complex data structures. This capability is particularly useful for handling nested data objects within grid columns. By default, the grid binds complex data to column fields using the dot (.) operator. However, when you render custom elements, such as input fields, in the edit template for a complex column, you must use the (___) underscore operator instead of the dot (.) operator to bind the complex object.
+
+In the following sample, the input element is rendered in the edit template of the FirstName and LastName column. The edited changes can be saved using the `name` property of the input element. Since the complex data is bound to the FirstName and LastName column, The `name` property should be defined as **Name___FirstName** and **Name___LastName**, respectively, instead of using the dot notation (**Name.FirstName** and **Name.LastName**).
```
[the remark this answers is not in the document]
Comment (harinikrh) [reply ej2-javascript/grid/editing/edit.md]: removed unecessary **

File: ej2-javascript/grid/editing/edit.md (hunk 3 above)
Comment (harinikrh) [reply ej2-javascript/grid/editing/edit.md:277]: isPrimaryKey is enabled

File: ej2-javascript/grid/editing/edit.md (hunk 4 above)
Comment (harinikrh) [reply ej2-javascript/grid/editing/edit.md]: changed the property name in to event

**Review by VigneshNatarajan27 — APPROVED** [review]
changes are fine

**Pull request merged**